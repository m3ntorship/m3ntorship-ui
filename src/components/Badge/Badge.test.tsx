import React from 'react';
import * as renderer from 'react-test-renderer';

import { Badge } from '.';

describe('Badge component', () => {
  it("should render empty badge label with primary background color when we don't pass any props", () => {
    const tree = renderer.create(<Badge />);
    expect(tree).toMatchSnapshot();
  });

  it('should render Badge component with text when we pass children', () => {
    const tree = renderer.create(<Badge>Simple badge</Badge>);
    expect(tree).toMatchSnapshot();
  });

  it('should render Badge component with text and Secondary background color when we pass children and color props', () => {
    const tree = renderer.create(<Badge color="secondary">Simple badge</Badge>);
    expect(tree).toMatchSnapshot();
  });

  it('should render Badge component with capitalized first letter', () => {
    const tree = renderer.create(<Badge>simple badge</Badge>);
    expect(tree).toMatchSnapshot();
  });
});
