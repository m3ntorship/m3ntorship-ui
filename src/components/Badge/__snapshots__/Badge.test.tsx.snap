// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component should render Badge component with capitalized first letter 1`] = `
<label
  className="shared variant-primary filled-state"
>
  Simple badge
</label>
`;

exports[`Badge component should render Badge component with text and Secondary background color when we pass children and color props 1`] = `
<label
  className="shared variant-secondary filled-state"
>
  Simple badge
</label>
`;

exports[`Badge component should render Badge component with text when we pass children 1`] = `
<label
  className="shared variant-primary filled-state"
>
  Simple badge
</label>
`;

exports[`Badge component should render empty badge label with primary background color when we don't pass any props 1`] = `
<label
  className="shared variant-primary empty-state"
/>
`;
